qcodes/instrument_drivers/tektronix/DPO7200xx.py

```diff
@@ -0,0 +1,605 @@
+        "MATH": "math",
+        "REF": "reference"
+    }
+
+    def __init__(
+            self,
+            parent: Instrument,
+            name: str,
+            channel_type: str,
+            channel_number: int,
+    ) -> None:
+
+        if channel_type not in self.channel_types:
+            acceptable_types = "".join(self.channel_types.keys())
+            raise ValueError(
+                f"Channel type needs to be one of {acceptable_types}"
+            )
+
+        super().__init__(parent, name)
+
+        self._channel_type = channel_type
+        self._channel_number = channel_number
+        self._identifier = f"{channel_type}{channel_number}"
+
+        waveform_format = _TektronixDPOWaveformFormat(
+            cast(VisaInstrument, self.parent),
+            "_waveform_format"
+        )
+
+        self.add_submodule(
+            "_waveform_format",
+            waveform_format
+        )
+
+        self.add_submodule(
+            "_data",
+            _TektronixDPOData(
+                cast(Instrument, self),
+                "_data"
+            )
+        )
+
+        self.add_parameter(
+            "scale",
+            get_cmd=f"{self._identifier}:SCA?",
+            set_cmd=f"{self._identifier}:SCA {{}}",
+            get_parser=float,
+            unit="V/div"
+        )
+
+        self.add_parameter(
+            "offset",
+            get_cmd=f"{self._identifier}:OFFS?",
+            set_cmd=f"{self._identifier}:OFFS {{}}",
+            get_parser=float,
+            unit="V"
+        )
+
+        self.add_parameter(
+            "position",
+            get_cmd=f"{self._identifier}:POS?",
+            set_cmd=f"{self._identifier}:POS {{}}",
+            get_parser=float,
+            unit="V"
+        )
+
+        self.add_parameter(
+            "termination",
+            get_cmd=f"{self._identifier}:TER?",
+            set_cmd=f"{self._identifier}:TER {{}}",
+            vals=Enum(50, 1E6),
+            get_parser=float,
+            unit="Ohm"
+        )
+
+        self.add_parameter(
+            "analog_to_digital_threshold",
+            get_cmd=f"{self._identifier}:THRESH?",
+            set_cmd=f"{self._identifier}:THRESH {{}}",
+            get_parser=float,
+            unit="V",
+        )
+
+        self.add_parameter(
+            "termination_voltage",
+            get_cmd=f"{self._identifier}:VTERm:BIAS?",
+            set_cmd=f"{self._identifier}:VTERm:BIAS {{}}",
+            get_parser=float,
+            unit="V"
+        )
+
+        self.add_parameter(
+            "trace_axis",
+            label="Time",
+            get_cmd=self._get_trace_setpoints,
+            vals=Arrays(shape=(self.get_trace_length,)),
+            unit=waveform_format.x_unit()
+        )
+
+        self.add_parameter(
+            "trace",
+            label="Voltage",
+            get_cmd=self._get_trace_data,
+            vals=Arrays(shape=(self.get_trace_length,)),
+            unit=waveform_format.y_unit(),
```

Comment:
perhaps a stupid question - does the unit have to be dynamic? so that if it changes, the unit of this parameter gets updated?

Reply:
not a stupid question at all! I modified my original PR so this is not relevant anymore, but the units on the channels will always be the same and there is no way for the user to change this. So this can be queried once.